File: Singleton/Singleton.java
package Singleton;

public class Singleton {
    static private Singleton color = null;
    private String bg;
    
    protected Singleton(){

    }

    public static Singleton Instance(){
        if (color == null){
            color = new Singleton();
        }
        return color;
    } 

    public void setBg(String bg){
        this.bg = bg;
    }

    public String getBg(){
        return this.bg;
    }
}

File: Singleton/Singleton.lua
Single = {}
Single.__index = Single


function Single:Instance()
  if not Single.instanced then
    local obj = setmetatable({}, self)
    obj.instanced = true
    obj.color = nil
    Single = obj
  end
  
  return Single
end

function Single:setColor(color)
  self.color = color
end

function Single:getColor()
  return color
end

inst = Single:Instance()
print(inst.color)
inst2 = Single:Instance()
inst:setColor("blue")
print(inst2.color)
